### ISSUE TITLE
Problem with bluetooth printer when printer is off, but paired with device 

### ISSUE THREAD

**Issue by AbdulBsit** [thiendangit/react-native-thermal-receipt-printer-image-qr#36]
**Problem**
While connecting with paired device, which is off, will hangs the whole App
I checked the JS thread its going in -2, which basically affecting performance

**How to reproduce**
Create a component that automatically connects to a pair bluetooth printer,
 but not turned on when the component mounts/renders.

**Possible Solution**
Connection timeout for BLEPrinter

**Comment by thiendangit** [comment]
@AbdulBsit tks your feedback! i'll try it when i have free time

**Comment by codoffer** [comment]
@thiendangit @AbdulBsit  I am facing the same issue with the paired device. I am saving the selected device to store and reconnect but the application is crashing

![Screenshot_20220311-135403](https://user-images.githubusercontent.com/16318845/157830365-78ce3cc9-a85d-4916-b14c-af59d3be7a51.jpg)
.

**Comment by AbdulBsit** [comment]
@codoffer This is happening because , you are not calling init before calling connect

**Comment by codoffer** [comment]
@AbdulBsit Yes, you may be right. I will call printer init() in print function on other screens.

I am having a single screen for device setup (Device Setting).  Using this screen, I have connected and printed the sample perfectly and saved the connected printer to the state (storage) to use the same connected printer on other screens. 

It's working fine till the printer is ON. After restarting the printer, It lost connection and gave the above error. 

I want to show the alert messages on printers off or paper jam etc. 

Thanks for your help

**Comment by AbdulBsit** [comment]
I have searched everywhere , i found there is no way to find the active status of device, only way is to try connecting the device, if promise failed, we assume device may be off or connected with other device, but here problem is, When you call connect function on off printer, it will hang for few seconds, may be because of the implementation of connect function is not handled correctly for this case, It is consuming JS Thread, which is affecting the rendering of UI, resulting in the unresponsive screen for few seconds
